modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -254,6 +259,48 @@ StorageOptions createStorageOptions(
         return storageOptionsBuilder.build();
     }
 
+    protected StorageRetryStrategy getRetryStrategy() {
+        return new RetryOnNetworkOutageRetryStrategy(StorageRetryStrategy.getLegacyStorageRetryStrategy());
+    }
+
+    private static class RetryOnNetworkOutageRetryStrategy implements StorageRetryStrategy {
+
+        private final ResultRetryAlgorithm<?> idempotentRetryAlgorithm;
+        private final ResultRetryAlgorithm<?> nonIdempotentRetryAlgorithm;
+
+        private RetryOnNetworkOutageRetryStrategy(StorageRetryStrategy delegate) {
+            this.idempotentRetryAlgorithm = new DelegatingResultRetryAlgorithm<>(delegate.getIdempotentHandler());
+            this.nonIdempotentRetryAlgorithm = new DelegatingResultRetryAlgorithm<>(delegate.getIdempotentHandler());
+        }
+
+        @Override
+        public ResultRetryAlgorithm<?> getIdempotentHandler() {
+            return idempotentRetryAlgorithm;
+        }
+
+        @Override
+        public ResultRetryAlgorithm<?> getNonidempotentHandler() {
+            return nonIdempotentRetryAlgorithm;
+        }
+
+        private record DelegatingResultRetryAlgorithm<T>(ResultRetryAlgorithm<T> delegate) implements ResultRetryAlgorithm<T> {
+
+            @Override
+            public TimedAttemptSettings createNextAttempt(Throwable prevThrowable, T prevResponse, TimedAttemptSettings prevSettings) {
+                return delegate.createNextAttempt(prevThrowable, prevResponse, prevSettings);
+            }
+
+            @Override
+            public boolean shouldRetry(Throwable prevThrowable, T prevResponse) throws CancellationException {
+                // Retry in the event of an unknown host exception
+                if (ExceptionsHelper.unwrap(prevThrowable, UnknownHostException.class) != null) {
+                    return true;
+                }
+                return delegate.shouldRetry(prevThrowable, prevResponse);
+            }
+        }
+    }
```

Comment:
A lot of code to decorate the `StorageRetryStrategy` but all the implementations are final and package private, and I think it's valuable to be able to delegate to one of the standard ones.